contracts/src/PDPService.sol

```diff
@@ -147,6 +154,29 @@ contract PDPService {
         return rootLeafCounts[setId][rootId];
     }
 
+    // Returns the last challenged leaf for a given proof set
+    function getLastChallengedLeaf(uint256 setId) public view returns (uint256) {
+        require(proofSetLive(setId), "Proof set not live");
+        return lastChallengedLeaf[setId];
+    }
+
+    // Returns the scheduled removals for a given proof set
```

Comment:
Returns the indices of roots that are scheduled for removal for the next proving period.